primary_beam_correction.py

```diff
@@ -207,63 +195,88 @@ def write_new_fits(pbc_image, path, outputFilename):
     hdr = images.header
     newhdr = hdr.copy()
     # add in required beam keywords
-    if newhdr['CLEANBMJ'] < 0:
-        # add in required beam keywords
-        newhdr['BMAJ'] = float(newhdr['HISTORY'][-2][20:30])
-        newhdr['BMIN'] = float(newhdr['HISTORY'][-2][36:48])
-        newhdr['BPA'] = float(newhdr['HISTORY'][-2][55:])
+    try:
+        if newhdr['CLEANBMJ'] < 0:
+            # add in required beam keywords
+            newhdr['BMAJ'] = float(newhdr['HISTORY'][-2][20:30])
+            newhdr['BMIN'] = float(newhdr['HISTORY'][-2][36:48])
+            newhdr['BPA'] = float(newhdr['HISTORY'][-2][55:])
 
-        newhdr['CTYPE3'] = 'FREQ'
-        newhdr['NAXIS3'] = 1
-        newhdr['CDELT3'] = 1.0
-        newhdr['CRVAL3'] = 1.284e9
-    else:
-        # add in required beam keywords
-        newhdr['BMAJ'] = newhdr['CLEANBMJ']
-        newhdr['BMIN'] = newhdr['CLEANBMN']
-        newhdr['BPA'] = newhdr['CLEANBPA']
-        # change the frequency plane keywords, we don't want multiple frequency
-        newhdr['CTYPE3'] = 'FREQ'
-        newhdr['NAXIS3'] = 1
-        newhdr['CDELT3'] = 1.0
+            newhdr['CTYPE3'] = 'FREQ'
+            newhdr['NAXIS3'] = 1
+            newhdr['CDELT3'] = 1.0
+            newhdr['CRVAL3'] = 1.284e9
+        else:
+            # add in required beam keywords
+            newhdr['BMAJ'] = newhdr['CLEANBMJ']
+            newhdr['BMIN'] = newhdr['CLEANBMN']
+            newhdr['BPA'] = newhdr['CLEANBPA']
+            # change the frequency plane keywords, we don't want multiple frequency
+            newhdr['CTYPE3'] = 'FREQ'
+            newhdr['NAXIS3'] = 1
+            newhdr['CDELT3'] = 1.0
```

Comment:
This is repeated from the main `if` statement above. Its always good practice to restore the happy path as soon as possible. So perhaps put these lines outside the `if ... else`. They should probably also go outside the `try ... except` as well since they don't ever do anything that you are trying to catch an exception for.

Reply:
That makes a lot of sense, i have fixed it also.